atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/persistentlock/LockStoreTest.java

```diff
@@ -0,0 +1,134 @@
+import static org.junit.Assert.assertThat;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyObject;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.atLeastOnce;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.verify;
+
+import java.util.Set;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import com.google.common.collect.ImmutableList;
+import com.palantir.atlasdb.AtlasDbConstants;
+import com.palantir.atlasdb.keyvalue.api.CheckAndSetException;
+import com.palantir.atlasdb.keyvalue.impl.InMemoryKeyValueService;
+
+public class LockStoreTest {
+    private static final String REASON = "reason";
+
+    private InMemoryKeyValueService kvs;
+    private LockStore lockStore;
+
+    @Before
+    public void setUp() throws Exception {
+        kvs = spy(new InMemoryKeyValueService(false));
+        lockStore = LockStore.create(kvs);
+    }
+
+    @Test
+    public void createsPersistedLocksTable() {
+        LockStore.create(kvs);
+        verify(kvs, atLeastOnce()).createTable(eq(AtlasDbConstants.PERSISTED_LOCKS_TABLE), any(byte[].class));
+    }
+
+    @Test
+    public void lockIsInitiallyOpen() {
+        Set<LockEntry> lockEntries = lockStore.allLockEntries();
+
+        assertThat(lockEntries, contains(LockStore.LOCK_OPEN));
+    }
+
+    @Test
+    public void noErrorIfLockOpenedWhileCreatingTable() {
+        doThrow(new CheckAndSetException("foo", null, null, ImmutableList.of())).when(kvs).checkAndSet(anyObject());
+
+        lockStore.populateTable(); // should not throw
+    }
+
+    @Test
+    public void canAcquireLock() throws Exception {
+        LockEntry lockEntry = lockStore.acquireLock(REASON);
+
+        assertThat(lockStore.allLockEntries(), contains(lockEntry));
+    }
+
+    @Test(expected = PersistentLockIsTakenException.class)
+    public void canNotAcquireLockTwice() throws Exception {
+        lockStore.acquireLock(REASON);
+        lockStore.acquireLock(REASON);
+    }
+
+    @Test(expected = PersistentLockIsTakenException.class)
+    public void canNotAcquireLockTwiceForDifferentReasons() throws Exception {
+        lockStore.acquireLock(REASON);
+        lockStore.acquireLock("other-reason");
+    }
+
+    @Test(expected = PersistentLockIsTakenException.class)
+    public void canNotAcquireLockThatWasTakenOutByAnotherStore() throws Exception {
+        LockStore otherLockStore = LockStore.create(kvs);
+        otherLockStore.acquireLock("grabbed by other store");
+
+        lockStore.acquireLock(REASON);
+    }
+
+    @Test
+    public void releaseLockPopulatesStoreWithOpenValue() throws Exception {
+        LockEntry lockEntry = lockStore.acquireLock(REASON);
+        lockStore.releaseLock(lockEntry);
+
+        assertThat(lockStore.allLockEntries(), contains(LockStore.LOCK_OPEN));
+    }
+
+    @Test(expected = PersistentLockIsTakenException.class)
```

Comment:
Yep, it's taken. But in general I see your point - going to remove `PersistentLockIsTakenException`. It was copied over from @gbrova's code before we thought of adding CheckAndSet and its associated exception.